### ISSUE TITLE
[Bug]: "Playwright Test for VSCode" v1.1.1 - Unable to "Execute Using Profile" anymore

### ISSUE THREAD

**Issue by mbouchardC3** [microsoft/playwright#30799]
### Version

Playwright: 1.42.1
Playwright Test for VSCode extension: 1.1.1

### Steps to reproduce

1. Go to "Testing" tab (of the Playwright extension)
2. Right-click on a spec file (test case)

### Expected behavior

The option "Execute Using Profile" is available and we can debug run the test case (which is the ONLY option that works to debug our project, since the usual "Run | Debug" options don't work for us

### Actual behavior

The option "Execute Using Profile" anymore. We instead have an option called "Start Continous Run Using"...
It's pretty bad, considering "Execute Using Profile" was the only way for us to debug our tests locally.
We now need to revert the extension version to continue working with it

Previous working version (1.0.22):
![image](https://github.com/microsoft/playwright/assets/125605193/c7b55dc5-555c-47ee-aff2-23fadcd8ebe1)
Latest broken version (1.1.1)
![image](https://github.com/microsoft/playwright/assets/125605193/38e6af2d-5335-45c0-9037-baa510bfa66d)

### Environment

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (16) x64 Intel(R) Core(TM) i9-10885H CPU @ 2.40GHz
    Memory: 31.36 GB / 63.75 GB
  Binaries:
    Node: 20.11.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.7.0 - C:\Program Files\nodejs\npm.CMD
  IDEs:
    VSCode: 1.89.1 - C:\Users\mbouchard\AppData\Local\Programs\Microsoft VS Code\bin\code.CMD
  npmPackages:
    @playwright/test: 1.42.1 => 1.42.1
```

**Comment by mxschmitt** [comment]
Have you tried selecting different config files using the new settings menu?
 
<img width="644" alt="Screenshot 2024-05-14 at 18 51 05" src="https://github.com/microsoft/playwright/assets/17984549/e2ce4a47-feae-45ae-99d1-a963472079b1">

**Comment by mbouchardC3** [comment]
> Have you tried selecting different config files using the new settings menu?
> 
> <img alt="Screenshot 2024-05-14 at 18 51 05" width="644" src="https://private-user-images.githubusercontent.com/17984549/330525974-e2ce4a47-feae-45ae-99d1-a963472079b1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU3MDY5NDIsIm5iZiI6MTcxNTcwNjY0MiwicGF0aCI6Ii8xNzk4NDU0OS8zMzA1MjU5NzQtZTJjZTRhNDctZmVhZS00NWFlLTk5ZDEtYTk2MzQ3MjA3OWIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE0VDE3MTA0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY0OTE2MmU5ZmYwZjkwYTlhYWFmZDNmMDczMWJiNTlmNDAwZThhN2Q0YTg5YzdkOTc1OGYyNTlmNWM0ZDhjYjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KKFjRYlbn2sbX6XGwjQ6-T8_F5L0G_ILwk4phCA3188">

I don't have any playwright.config.ts under my Playwright tab here
![image](https://github.com/microsoft/playwright/assets/125605193/c366c289-5707-413d-a860-33fd17539b0d)

**Comment by yury-s** [comment]
> I don't have any playwright.config.ts under my Playwright tab here

Would it be possible for you to share the project where it happens?

**Comment by mbouchardC3** [comment]
> > I don't have any playwright.config.ts under my Playwright tab here
> 
> Would it be possible for you to share the project where it happens?

I'd rather not share the project, as it has company intellectual property in it.

Any specific information you need that I could try to get for you?

**Comment by skonakalla455** [comment]
I see even the older version doesn't work. I reverted it to older version 1.0.22 and licked the TRAINGLE icon to run/debug but it said "The test run did not record any output. "
What is the other alternative to debug my playwright type script test.

**Comment by trajchelazovski** [comment]
@mbouchardC3 I experienced the same initial confusion as you after upgrading to the latest version of the Playwright Test for VSCode extension.

In my case I had multiple projects defined and in order for the tests to show up in the files and test tab and the icons in the files you need to select the projects (previously called Default Profile).
![image](https://github.com/microsoft/playwright/assets/15001723/eccda1dc-dc14-4b39-bf5a-4b764ccb3caa)

I personally found it better as it was before since it allowed you to change on the fly the project.
![image](https://github.com/microsoft/playwright/assets/15001723/77623c91-2413-46b0-86e3-5282262ed3ca)

**Comment by mbouchardC3** [comment]
> I see even the older version doesn't work. I reverted it to older version 1.0.22 and clicked the TRAINGLE icon to run/debug but it said "The test run did not record any output. " Can anyone suggest What is the other alternative to debug my playwright type script test?

As mentioned in my initial post, I have the same problem as you. My alternate solution was to right click on the test and click on the "Execute Using Profile" option. Doing it from there works for us, but this option doesn't exist anymore in 1.1.1.

**Comment by mbouchardC3** [comment]
> @mbouchardC3 I experienced the same initial confusion as you after upgrading to the latest version of the Playwright Test for VSCode extension 1.1.1.
> 
> In my case I had multiple projects defined and in order for the tests to show up in the files and test tab and the icons in the files you need to select the projects (previously called Default Profile). ![image](https://private-user-images.githubusercontent.com/15001723/330754653-eccda1dc-dc14-4b39-bf5a-4b764ccb3caa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU3ODE2NzcsIm5iZiI6MTcxNTc4MTM3NywicGF0aCI6Ii8xNTAwMTcyMy8zMzA3NTQ2NTMtZWNjZGExZGMtZGMxNC00YjM5LWJmNWEtNGI3NjRjY2IzY2FhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE1VDEzNTYxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAzY2IwZmYyNjk2NTgwOGYyNzJkN2VlYTk3NGYzMzM3OGUyYWY5NTI3OTU5YWRmMWVmYjA1YmNkNDEwYjM0OGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oxr42OJFbtYOL1syMHm9if8Nw4atViRlb7CJi9EW6VI)
> 
> I personally found it better as it was before (in version 1.0.22) since it allowed you to change on the fly the project. ![image](https://private-user-images.githubusercontent.com/15001723/330756105-77623c91-2413-46b0-86e3-5282262ed3ca.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU3ODE2NzcsIm5iZiI6MTcxNTc4MTM3NywicGF0aCI6Ii8xNTAwMTcyMy8zMzA3NTYxMDUtNzc2MjNjOTEtMjQxMy00NmIwLTg2ZTMtNTI4MjI2MmVkM2NhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE1VDEzNTYxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMwYjg5NzY3YTFmZTFmNTY1MzczMThkNjkyMmQ4YjYzYjZmMGQwYzZkOTQwOWM4ZWUzYjRhZGE2NGYzOWUzMjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.RQFAjsmsLnJjdVgOjVV4CagsjICUrSJ6J0LcYWZKa9k)

I know about this settings change and I welcome it. However, choosing my projects like so only allows me to see them in my test explorer. It doesn't fix my problem that I cannot run/debug my tests locally like I could in the previous version
![image](https://github.com/microsoft/playwright/assets/125605193/cbe69218-9823-4a6b-a017-c5f0b8cac986)

**Comment by yury-s** [comment]
> I'd rather not share the project, as it has company intellectual property in it.
> 
> Any specific information you need that I could try to get for you?

We need to understand why in your configuration with multiple playwright configs there is no cogwheel button that allows to select which configs to use. The problem will likely persist if you delete all sensitive stuff from your project and leave just the directory structure and dummy test files.

**Comment by mbouchardC3** [comment]
Do you need a screenshot from my project route? A sample of my playwright.config.ts?
I'll provide what is needed and filter out company IP if applicable

**Comment by yury-s** [comment]
> Do you need a screenshot from my project route?

No, we cannot debug screenshots.

> A sample of my playwright.config.ts? I'll provide what is needed and filter out company IP if applicable

My understanding of https://github.com/microsoft/playwright/issues/30799#issuecomment-2110735785 was that you have multiple playwright.config.ts in your project but you didn't have the cogwheel button like [this](https://github.com/microsoft/playwright/issues/30799#issuecomment-2110696456). If that's the case, we need a minimal subset of your project files where this reproduces (including configs, tests etc).

> However, choosing my projects like so only allows me to see them in my test explorer. It doesn't fix my problem that I cannot run/debug my tests locally like I could in the previous version

Can you elaborate on why you cannot debug your tests in this case? If you select the only project that you need in the sidebar and then click debug on a spec file/test case, does it not work or you want something else?

**Comment by mbouchardC3** [comment]
> > Do you need a screenshot from my project route?
> 
> No, we cannot debug screenshots.
> 
> > A sample of my playwright.config.ts? I'll provide what is needed and filter out company IP if applicable
> 
> My understanding of [#30799 (comment)](https://github.com/microsoft/playwright/issues/30799#issuecomment-2110735785) was that you have multiple playwright.config.ts in your project but you didn't have the cogwheel button like [this](https://github.com/microsoft/playwright/issues/30799#issuecomment-2110696456). If that's the case, we need a minimal subset of your project files where this reproduces (including configs, tests etc).
> 
> > However, choosing my projects like so only allows me to see them in my test explorer. It doesn't fix my problem that I cannot run/debug my tests locally like I could in the previous version
> 
> Can you elaborate on why you cannot debug your tests in this case? If you select the only project that you need in the sidebar and then click debug on a spec file/test case, does it not work or you want something else?

Selecting only one project in the sidebar and clicking on the debug button does not work on my end. The test run automatically stops and gives me the error "The test run did not record any output."

I know you can't use screenshots for troubleshooting, but here's my project tree anyway
![image](https://github.com/microsoft/playwright/assets/125605193/f265c06a-9650-4ec8-a784-1d047900df6f)

And here's my playwright.config.ts file
![image](https://github.com/microsoft/playwright/assets/125605193/b131c540-58dd-4806-b530-3427b1515fae)

To resume, I have test cases that I store in a folder and in another folder, I have my test suites that uses those exported test cases. Here's an example of one of my test suites:
![image](https://github.com/microsoft/playwright/assets/125605193/ba222f50-6716-4fbd-a51a-58cc2703a745)

The suite is able to use the test case, since we export it like so:
![image](https://github.com/microsoft/playwright/assets/125605193/d554883a-373a-498e-94c7-bfd206a937a1)

And instead of using the test option from Playwright, we use a test extender from our global-before-after-each class
![image](https://github.com/microsoft/playwright/assets/125605193/2eeead9d-1201-4268-9125-5c3dd2ad9602)

I blurred company info, but hopefully, this will be enough to help you understanding the issue. Thank you for taking time for this!

**Comment by skonakalla455** [comment]
We have a new version of Playwright Test for VSCODE 1.1.2 and it looks like it broke the older versions as well.
Old versions: Doesn't allow to debug
Latest version: Doesn't allow to debug

**Comment by yury-s** [comment]
@mbouchardC3 the message "Execute Playwright Suite" (and  "Execute Playwright Suite") comes from some other extension, not Playwright.

**Comment by mbouchardC3** [comment]
> @mbouchardC3 the message "Execute Playwright Suite" (and "Execute Playwright Suite") comes from some other extension, not Playwright.

I know. That's not what my issue is about. I don't use this "Execute Playwright Suite" snippet

**Comment by yury-s** [comment]
We compiled an [FAQ](https://github.com/microsoft/playwright/issues/30882) explaining the recent changes in the VS Code Extension, it doesn't directly address the issue in this bug, but nevertheless may be helpful for better understanding the new approach to selecting different configurations.

> Selecting only one project in the sidebar and clicking on the debug button does not work on my end. The test run automatically stops and gives me the error "The test run did not record any output."

Reading the entire thread again, I believe this is the fundamental issue and we need to focus on understanding why it's not working for you. @mbouchardC3, can you reduce your project to a minimal example that doesn't contain sensitive internal code, and share it with us, so that we can reproduce it locally and see why it's not working as intended?

**Comment by mbouchardC3** [comment]
@yury-s Sorry for the delay in response, long weekend here.

Anyway, after the release of the 1.1.5 extension AND upgrading playwright/test to latest, it works correctly now, even when I select all my projects!

If some people still struggle with this problem, try all latest versions (and don't forget to npx playwright install afterwards) to see if it works for you.

Thank you guys for the support!


TL;DR : The removal of the option "Execute Using Profile" is normal. Current latest versions (playwright/test 1.44.0 combined with Playwright Test for VS Code 1.1.5) work well together
